src/core/deltafilewrapper.h

```diff
@@ -378,6 +378,10 @@ class DeltaFileWrapper : public QObject
      */
     bool applyDeltasOnLayers( QHash<QString, QgsVectorLayer *> &vectorLayers, bool shouldApplyInReverse );
 
+    /**
+     * Add file checksums from relevant changed attributes.
+     */
+    void addAttachments( const QString &localLayerId, const QJsonObject &oldAttrs, const QJsonObject &newAttrs, QJsonObject &oldFileChecksums, QJsonObject &newFileChecksums );
```

Comment:
```suggestion
    /**
     * Add file checksums from relevant changed attributes.
     * \returns a std::tuple<oldChecksums, newChecksums>
     */
    void addAttachments( const QString &localLayerId, const QJsonObject &oldAttrs, const QJsonObject &newAttrs, QJsonObject &oldFileChecksums, QJsonObject &newFileChecksums );
```

We are happy to live in modern days where we no longer have to hack return values through references.